src/snapred/backend/recipe/algorithm/LiteDataCreationAlgo.py

```diff
@@ -0,0 +1,99 @@
+import json
+
+from mantid.api import *
+from mantid.api import AlgorithmFactory, PythonAlgorithm, mtd
+from mantid.kernel import Direction
+
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "LiteDataCreationAlgo"
+
+
+class LiteDataCreationAlgo(PythonAlgorithm):
+    def PyInit(self):
+        self.declareProperty("InputWorkspace", defaultValue="", direction=Direction.Input)
+        self.declareProperty("OutputWorkspace", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def PyExec(self):
+        self.log().notice("Lite Data Creation START!")
+        # define super pixel dimensions
+        superPixel = [8, 8]
+
+        # get output workspace name
+        outputWorkspaceName = self.getProperty("OutputWorkspace").value
+
+        # load input workspace
+        inputWorkspace = self.getProperty("InputWorkspace").value
+        self.mantidSnapper.mtd[inputWorkspace]
+
+        # clone the workspace
+        self.mantidSnapper.CloneWorkspace(
+            "Cloning input workspace for lite data creation...",
+            InputWorkspace=inputWorkspace,
+            OutputWorkspace=outputWorkspaceName,
+        )
+        self.mantidSnapper.executeQueue()
+        outputWorkspace = self.mantidSnapper.mtd[outputWorkspaceName]
+
+        # IEventWorkspace ID is equivalent to spec ID
+        numSpec = outputWorkspace.getNumberHistograms()
+
+        # array to hold spectrum data
+        spectrumData = []
+
+        # iterate over IEventWorkspace IDs
+        for spec in range(numSpec):
+            # get event list per workspace
+            spectrum = outputWorkspace.getSpectrum(workspaceIndex=spec)
+            spectrumData.append(spectrum)
+
+        # allocate array of event IDs
+        for spectrum in spectrumData:
+            eventIDs = spectrum.getDetectorIDs()
+            for event in eventIDs:
+                superID = self.getSuperID(nativeID=event, xdim=superPixel[0], ydim=superPixel[0])
+
+                # clear detector IDs and then add the new modified super ID
+                spectrum.clearDetectorIDs()
+                spectrum.addDetectorID(superID)
+
+        self.mantidSnapper.LoadInstrument(
+            "Loading instrument...",
+            Workspace=outputWorkspace,
+            Filename="/SNS/SNAP/shared/Calibration/Powder/SNAPLite.xml",
+            RewriteSpectraMap=False,
+        )
+        # self.mantidSnapper.DeleteWorkspace("Cleaning up input workspace...", Workspace=ws)
+        self.mantidSnapper.executeQueue()
+
+        self.setProperty("OutputWorkspace", outputWorkspaceName)
+        return outputWorkspaceName
+
+    def getSuperID(self, nativeID, xdim, ydim):
+        # Constants
+        Nx, Ny = 256, 256  # Native number of horizontal and vertical pixels
```

Comment:
My same concern about hard-coded values.

Reply:
Again, no instance of this within the instrument definition file.

<?xml version='1.0' encoding='UTF-8'?>
<!-- For help on the notation used to specify an Instrument Definition File
     see http://www.mantidproject.org/IDF -->
<instrument xmlns="http://www.mantidproject.org/IDF/1.0"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://www.mantidproject.org/IDF/1.0 http://schema.mantidproject.org/IDF/1.0/IDFSchema.xsd"
            name="SNAP" valid-from   ="2018-05-01 00:00:01"
                        valid-to     ="2100-01-31 23:59:59"
		        last-modified="2019-05-03 11:30:00">
  <!--Data taken from /SNS/SNAP/2010_2_3_CAL/calibrations/SNAP_geom_2010_03_22.xml-->
  <!--THIS IS A FAKE INSTRUMENT DEFINITION FOR TESTING ONLY-->
  <!--DEFAULTS-->
  <defaults>
    <length unit="metre"/>
    <angle unit="degree"/>
    <reference-frame>
      <along-beam axis="z"/>
      <pointing-up axis="y"/>
      <handedness val="right"/>
      <theta-sign axis="x"/>
    </reference-frame>
  </defaults>

  <!--SOURCE-->
  <component type="moderator">
    <location z="-15.0"/>
  </component>
  <type name="moderator" is="Source"/>

  <!--SAMPLE-->
  <component type="sample-position">
    <location y="0.0" x="0.0" z="-5.0"/>
  </component>
  <type name="sample-position" is="SamplePos"/>

  <!--MONITORS-->
  <idlist idname="only-monitor">
    <id val="-1"/>
  </idlist>
  <component type="only-monitor" idlist="only-monitor">
    <location/>
  </component>
  <type is="monitor" name="only-monitor">
    <component type="monitor">
      <location x="0.0" y="0.0" z="15" name="monitor1"/>
    </component>
  </type>

  <!--DETECTORS-->
  <component type="East">
    <location >
      <parameter name="roty">
        <logfile id="det_arc2" eq="180.0+value"/>
      </parameter>
      <parameter name="r-position">
        <logfile id="det_lin2" eq="0.5+value" />
      </parameter>
      <parameter name="t-position">
        <logfile id="det_arc2" />
      </parameter>
    </location>
  </component>

  <type name="East">
    <component type="Column1">
      <location/>
    </component>
  </type>

  <type name="Column1">
    <component type="panel" idstart="0" idfillbyfirst="y" idstepbyrow="1" >
      <location name="bank1">  <!-- was bank4 - 655360 -->
        <trans x="0.0" y="1.0" />
      </location>
    </component>
    <component type="panel" idstart="8" idfillbyfirst="y" idstepbyrow="1" >
      <location name="bank2">  <!-- was bank4 - 655360 -->
        <trans x="1.0" y="0.0" />
      </location>
    </component>
  </type>

<!-- Rectangular Detector Panel -->
<type name="panel" is="rectangular_detector" type="pixel"
    xpixels="1" xstart="0.0" xstep="+0.001"
    ypixels="8" ystart="0.0" ystep="+0.001" >
  <properties/>
</type>

<!-- Pixel for Detectors-->
<type is="detector" name="pixel">
  <cuboid id="pixel-shape">
    <left-front-bottom-point  y="0.0" x="0.0" z="0.0"/>
    <left-front-top-point     y="1.0" x="0.0" z="0.0"/>
    <left-back-bottom-point   y="0.0" x="0.0" z="-0.0001"/>
    <right-front-bottom-point y="0.0" x="1.0" z="0.0"/>
  </cuboid>
  <algebra val="pixel-shape"/>
</type>

<!-- Shape for Monitors-->
<!-- TODO: Update to real shape -->
<type is="monitor" name="monitor">
  <cylinder id="some-shape">
    <centre-of-bottom-base p="0.0" r="0.0" t="0.0"/>
    <axis y="0.0" x="0.0" z="1.0"/>
    <radius val="1.00"/>
    <height val="1.00"/>
  </cylinder>
  <algebra val="some-shape"/>
</type>

</instrument>

Reply:
You quoted a fake instrument I made the unit testing. But, it does get at what I needed.  If you look at the original [SNAP definition](https://github.com/neutrons/SNAPRed/blob/next/tests/resources/inputs/pixel_grouping/SNAP_Definition.xml), it has the following:
``` xml
<!-- Rectangular Detector Panel -->
<type name="panel" is="rectangular_detector" type="pixel"
    xpixels="256" xstart="-0.078795" xstep="+0.000618"
    ypixels="256" ystart="-0.078795" ystep="+0.000618" >
  <properties/>
</type>
```
and then in the [SNAPLite definition](https://github.com/neutrons/SNAPRed/blob/next/tests/resources/inputs/pixel_grouping/SNAPLite_Definition.xml), this has been reduced to:
``` xml
<!-- Rectangular Detector Panel -->
<type name="panel" is="rectangular_detector" type="pixel"
    xpixels="32" xstart="-0.076632" xstep="+0.004944"
    ypixels="32" ystart="-0.076632" ystep="+0.004944" >
  <properties/>
</type>

```
where 256/32 = 8.  So both of these numbers could be derived from instrument definitions.  Lemme check with @mguthriem if there is any reason why that might ever be desirable.

Reply:
Okay! Gotcha. Yeah, I'm unsure in a practicle sense that these values would change as the instrument's physical parameters won't change unless new upgraded detectors are used or different grouping methods are employed. Regardless, this is still hard coded but removed by an extra level of abstraction. You could point to these values within the instrument definition file, in specific, the local one (fakeSNAPLite.xml) but this would require a refactor of the instrument defintion file to update those values in case some fundamental change occurs to the instrument.

Reply:
@mguthriem suggests the use of lite mode is going to be specific to SNAP, so this is probably not worth worrying over at the moment.